core/src/main/java/org/apache/iceberg/hadoop/HadoopFileIO.java

```diff
@@ -211,6 +213,27 @@ private ExecutorService executorService() {
     return executorService;
   }
 
+  private void deletePath(FileSystem fs, Path toDelete, boolean recursive) throws IOException {
+    if (!shouldUseTrashIfConfigured()) {
+      fs.delete(toDelete, recursive);
+      return;
+    }
+    Trash trash = new Trash(fs, getConf());
+    if (!trash.isEnabled()) {
+      fs.delete(toDelete, recursive);
+    } else {
+      if (!recursive && fs.getFileStatus(toDelete).isDirectory()) {
```

Comment:
I feel like it would be better to not perform the check and allow a failure from the HDFS/library to propagate up.  The FileIO doesn't expose directory operations and we don't delete directories, so I think this just introduces and unnecessary call.